### PR TITLE
Add /help slash command

### PR DESCRIPTION
**Pull request by kkamdooong**
#### Summary
Add /help slash command

#### Ticket Link
[PLT-6715](https://mattermost.atlassian.net/browse/PLT-6715)

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates

### CODE REVIEW

File: api/command_help_test.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	"testing"
+	"github.com/mattermost/platform/utils"
+)
+
+func TestHelpCommand(t *testing.T) {
+	th := Setup().InitBasic()
+	Client := th.BasicClient
+	channel := th.BasicChannel
+
+	rs := Client.Must(Client.Command(channel.Id, "/help ")).Data.(*model.CommandResponse)
```
Comment (cpanato) [inline api/command_help_test.go]: I think the `Client.Must` will be deprecated soon, try to use the client for the v2 api

File: app/command_help.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
```
Comment (cpanato) [inline app/command_help.go]: should be 2017-present

File: i18n/en.json [hunk 3]
```diff
@@ -824,6 +824,14 @@
     "translation": "shrug"
   },
   {
+    "id": "api.command_help.desc",
+    "translation": "Open the Mattermost help page"
```
Comment (cpanato) [inline i18n/en.json]: Will this command open the help page or send the link to the user? and then the user open the link?

File: api/command_help_test.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
```
Comment (cpanato) [inline api/command_help_test.go]: should be 2017-present

File: api/command_help_test.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+func TestHelpCommand(t *testing.T) {
+	th := Setup().InitBasic()
+	Client := th.BasicClient
+	channel := th.BasicChannel
+
+	rs := Client.Must(Client.Command(channel.Id, "/help ")).Data.(*model.CommandResponse)
+	if *utils.Cfg.SupportSettings.HelpLink != rs.GotoLocation {
+		t.Fatal("failed to get help link")
+	}
```
Comment (cpanato) [inline api/command_help_test.go]: maybe can add a test when the help link is not set and return that.

File: api/command_help_test.go (hunk 1 above)
Comment (kkamdooong) [inline api/command_help_test.go]: @cpanato 
what does 'v2 api' mean?
is it Client4.executeCommand() what you said?

Comment (cpanato) [inline api/command_help_test.go]: I guess so

File: api/command_help_test.go (hunk 5 above)
Comment (kkamdooong) [inline api/command_help_test.go]: if help link is not set, `SUPPORT_SETTINGS_DEFAULT_HELP_LINK` is set to default.
https://github.com/mattermost/platform/blob/master/model/config.go#L791

File: i18n/en.json (hunk 3 above)
Comment (kkamdooong) [inline i18n/en.json]: In JIRA ticket, /help command open the help page.
goto location redirect user to webpage

File: api/command_help_test.go (hunk 1 above)
Comment (kkamdooong) [inline api/command_help_test.go]: add test [api4/command_help_test.go](https://github.com/mattermost/platform/pull/6650/files#diff-56095adc1437bba0f90a5b5b79fa6cab)

File: api/command_help_test.go (hunk 5 above)
Comment (cpanato) [inline api/command_help_test.go]: yes, just to check that, but maybe is too much :)

File: i18n/en.json (hunk 3 above)
Comment (cpanato) [inline i18n/en.json]: perfect then :)

**Comment by cpanato** [comment]
@jasonblais I did a smoke test and looks fine :)

**Comment by jasonblais** [comment]
Excellent work @kkamdooong, thank you for the contribution!

I tested it and found two small issues. Can you help take a look?

1 - Typing `/help` on a reply thread refreshes the page. Repro steps:
 - Post a message in any channel
 - Click the reply arrow beside the message, or click [...], then "Reply"
 - In the reply thread on the right-hand side, post `/help`

Observed: Page refreshes
Expected: Mattermost help page is opened

2 - If the [Help link](https://docs.mattermost.com/administration/config-settings.html#help-link) is set to blank, `/help` does nothing. Repro steps:
 - Go to System Console > Legal and Support > Help Link
 - Set the field to blank (i.e. no value), then hit "Save" at the bottom of the page
 - Go back to a Mattermost channel and post `/help`

Observed: Nothing happens
Expected: Open the default page (https://about.mattermost.com/default-help/). Note that the "Help" link should still be hidden from the Mattermost login page, sign-up pages, and Main Menu for this case.

@kkamdooong let me know if the second point complicates the code too much. Happy to discuss a simpler alternative.

**Comment by kkamdooong** [comment]
@jasonblais 
1. I guess this is webapp issue. excuteCommand logic is diffrent in create_post.jsx and create_comment.jsx. I make it to be equal.
2. I modified helpCommand to redirect default help link, when help link is ""(empty string).

**Comment by jasonblais** [comment]
Thanks!! Setting up a test server now

File: api/command_help_test.go [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+	"github.com/mattermost/platform/utils"
+	"testing"
+)
+
+func TestHelpCommand(t *testing.T) {
+	th := Setup().InitBasic()
+	Client := th.BasicClient
+	channel := th.BasicChannel
+
+	*utils.Cfg.SupportSettings.HelpLink = ""
```
Comment (hmhealey) [inline api/command_help_test.go]: Can you add a `defer` statement above this to return the setting to its previous value after the test? `utils.Cfg` persists between tests and its written back to config.json, so we want to keep it the same between runs. The look something like this (ripped from `api/emoji_test.go` if you want to see more examples):
```go
	EnableCustomEmoji := *utils.Cfg.ServiceSettings.EnableCustomEmoji
	defer func() {
		*utils.Cfg.ServiceSettings.EnableCustomEmoji = EnableCustomEmoji
	}()
	*utils.Cfg.ServiceSettings.EnableCustomEmoji = true
```

File: api4/command_help_test.go [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+	"github.com/mattermost/platform/model"
+	"github.com/mattermost/platform/utils"
+	"testing"
+)
+
+func TestHelpCommand(t *testing.T) {
+	th := Setup().InitBasic()
+	Client := th.Client
+	channel := th.BasicChannel
+
```
Comment (hmhealey) [inline api4/command_help_test.go]: Same comment as above

**Comment by kkamdooong** [comment]
@hmhealey 
I learned good thing because of you. I added c460379

**Comment by jasonblais** [comment]
@hmhealey Should be ready for review

**Pull request merged**